src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/Telemetry/LanguageServiceTelemetryService.cs

```diff
@@ -0,0 +1,100 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System.ComponentModel.Composition;
+using Microsoft.VisualStudio.ProjectSystem.LanguageServices;
+
+namespace Microsoft.VisualStudio.Telemetry
+{
+    [Export(typeof(ILanguageServiceTelemetryService))]
+    internal class LanguageServiceTelemetryService : ILanguageServiceTelemetryService
+    {
+        private readonly ITelemetryService _telemetryService;
+
+        [ImportingConstructor]
+        public LanguageServiceTelemetryService(ITelemetryService telemetryService)
+        {
+            _telemetryService = telemetryService;
+        }
+
+        /// <summary>
+        /// Hashes personally identifiable information for telemetry consumption.
+        /// </summary>
+        /// <param name="value">Value to hashed.</param>
+        /// <returns>Hashed value.</returns>
+        public string HashValue(string value)
+        {
+            return _telemetryService.HashValue(value);
+        }
+
+        /// <summary>
+        /// Post a telemetry event when a design time build fails.
+        /// </summary>
+        /// <param name="projectId">A unique identifier for the project.</param>
+        public void PostDesignTimeBuildFailureEvent(string projectId)
+        {
+            _telemetryService.PostProperties(TelemetryEventName.DesignTimeBuildComplete, new (string propertyName, object propertyValue)[]
+                {
+                    ( TelemetryPropertyName.DesignTimeBuildCompleteSucceeded, false),
```

Comment:
This `false` will be boxed. Might be good to reuse a boxed value by assigning it to a field.

CPS has an internal `Boxed` class for such values. Maybe we create one in this repo too.